### PR TITLE
Add confirmation modal when deleting/ regenerating api keys, deleting webhook

### LINKED ISSUE #4016: Add a confirmation modal when deleting/regenerating an API Key
Currently you can invalidate an API Key and break an entire application by clicking "Regenerate key" or "Delete".
This is dangerous and it'd be nice to have a confirmation message to prevent any error.

In account deleting we already have a mechanism to confirm the action:

Screenshot 2024-02-16 at 15 46 54

We should implement the same mechanism for token regeneration and token deletion.

The message should be 
`Please type "yes" to confirm you want to delete this API Key. 
Be aware that any script using this key will stop working. `

`If you’ve lost this key, you can regenerate it, but be aware that any script using this key will need to be updated.
Please type "yes" to confirm`

### PR DESCRIPTION
**Pull request by anoopw3bdev**
Closes : #4016 

This PR add a confirmation modal when deleting/ regenerating api keys or when deleting a webhook.

This also adds a little more spacing between the text and input field, swap the position of "Cancel" and "Delete" button in the confirmation modal as per [this](https://github.com/twentyhq/twenty/issues/4016#issuecomment-1948930169) discussion 
commit: 7098117cab5b1555087390563aabc893e1f1da96

### CODE REVIEW

**Comment by charlesBochet** [comment]
Thanks @anoopw3bdev, I have nothing to say on the code part :)

Could you replace the "Yes" placeholder by "yes" to be consistent with the modal message?

File: packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx [hunk 1]
```diff
@@ -186,12 +191,43 @@ export const SettingsDevelopersApiKeyDetail = () => {
...
+            Please type "yes" to confirm you want to delete this API Key. Be
+            aware that any script using this key will stop working.
+          </>
+        }
+        onConfirmClick={deleteIntegration}
+        deleteButtonText="Delete"
+      />
+      <ConfirmationModal
+        confirmationPlaceholder="Yes"
+        confirmationValue="Yes"
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx:217]:
```suggestion
        confirmationValue="yes"
```

File: packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx [hunk 2]
```diff
@@ -186,12 +191,43 @@ export const SettingsDevelopersApiKeyDetail = () => {
...
-                onClick={() => deleteIntegration()}
+                onClick={() => setIsDeleteApiKeyModalOpen(true)}
               />
             </Section>
           </SettingsPageContainer>
         </SubMenuTopBarContainer>
       )}
+      <ConfirmationModal
+        confirmationPlaceholder="Yes"
+        confirmationValue="Yes"
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx:202]:
```suggestion
        confirmationValue="yes"
```

File: packages/twenty-front/src/pages/settings/developers/webhooks/SettingsDevelopersWebhookDetail.tsx [hunk 3]
```diff
@@ -62,7 +66,22 @@ export const SettingsDevelopersWebhooksDetail = () => {
                 variant="secondary"
                 title="Disable"
                 Icon={IconTrash}
-                onClick={() => deleteWebhook()}
+                onClick={() => setIsDeleteWebhookModalOpen(true)}
+              />
+              <ConfirmationModal
+                confirmationPlaceholder="Yes"
+                confirmationValue="Yes"
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/developers/webhooks/SettingsDevelopersWebhookDetail.tsx:73]:
```suggestion
                confirmationValue="yes"
```

File: packages/twenty-front/src/pages/settings/developers/webhooks/SettingsDevelopersWebhookDetail.tsx [hunk 4]
```diff
@@ -62,7 +66,22 @@ export const SettingsDevelopersWebhooksDetail = () => {
                 variant="secondary"
                 title="Disable"
                 Icon={IconTrash}
-                onClick={() => deleteWebhook()}
+                onClick={() => setIsDeleteWebhookModalOpen(true)}
+              />
+              <ConfirmationModal
+                confirmationPlaceholder="Yes"
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/developers/webhooks/SettingsDevelopersWebhookDetail.tsx:72]:
```suggestion
                confirmationPlaceholder="yes"
```

File: packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx [hunk 5]
```diff
@@ -186,12 +191,43 @@ export const SettingsDevelopersApiKeyDetail = () => {
...
+          <>
+            Please type "yes" to confirm you want to delete this API Key. Be
+            aware that any script using this key will stop working.
+          </>
+        }
+        onConfirmClick={deleteIntegration}
+        deleteButtonText="Delete"
+      />
+      <ConfirmationModal
+        confirmationPlaceholder="Yes"
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx:216]:
```suggestion
        confirmationPlaceholder="yes"
```

File: packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx [hunk 6]
```diff
@@ -186,12 +191,43 @@ export const SettingsDevelopersApiKeyDetail = () => {
...
                 Icon={IconTrash}
-                onClick={() => deleteIntegration()}
+                onClick={() => setIsDeleteApiKeyModalOpen(true)}
               />
             </Section>
           </SettingsPageContainer>
         </SubMenuTopBarContainer>
       )}
+      <ConfirmationModal
+        confirmationPlaceholder="Yes"
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx:201]:
```suggestion
        confirmationPlaceholder="yes"
```

**Review by charlesBochet — APPROVED** [review]
I've made the "Yes" to "yes" update myself so you don't lose your time for such a small change.
Thank you!

**Comment by anoopw3bdev** [comment]
Thank you @charlesBochet for reviewing and correcting the text :)

**Pull request merged**